tests/NCronJob.Tests/NCronJobIntegrationTests.cs

```diff
@@ -204,22 +204,6 @@ public async Task LongRunningJobShouldNotBlockScheduler()
         jobFinished.ShouldBeTrue();
     }
 
-    [Fact]
-    public async Task ThrowIfJobWithDependenciesIsNotRegistered()
```

Comment:
This is partially covered by the other test I added. Partially, because we just don't blow up anymore anyway.

That is somewhat of a major change. Earlier NCronJob just "exploded" in this case. But to be honest, I don't like it. Because this can blow up after days, hours or weeks and not in the beginning. So we should handle this anyway more gracefully.

Thoughts @nulltoken

Reply:
I was thinking about leveraging UseNCronJobAsync to ensure the resolvability of every registered job *before* startup.

Reply:
Good point. I am not sure if this is our responsibility in the first place.

Microsofts Abstraction in ASP.NET already has the means to do so:

```csharp
builder.Host.UseDefaultServiceProvider((_, options) =>
{
    options.ValidateScopes = true;
    options.ValidateOnBuild = true;
});
```

So maybe this is more of a documentation issue rather than we do this (again).

As we don't have access to internals, it might get quite expensive to check. Mainly because we have to instantiate many services (and check for the correct scope).

Reply:
On a second thought - `builder.Host.UseDefaultServiceProvider` will probably not help here, as we do everything at runtime and nothing at compile time.

But the second argument still stands - it might be a very expensive thing to do.
Maybe we can have options as well, that would enable checking.

Reply:
In any case - I'd make it a follow up task.

Reply:
How about skipping it rather than drop it completely?
It would remind us that something in this area has to somehow be covered.